File: contents/bubble_sort/code/c++/bubblesort.cpp
#include <algorithm>
#include <cstddef>
#include <iostream>
#include <iterator>

template <class Iter>
void print_range(Iter first, Iter last) {
  for (auto it = first; it != last; ++it)
    std::cout << *it << " ";
  std::cout << std::endl;
}

template <class Iter>
void bubble_sort(Iter first, Iter last) {
  for (auto it1 = first; it1 != last; ++it1) {
    for (auto it2 = first; it2 + 1 != last; ++it2) {
      // these are unsorted! gotta swap 'em
      if (*(it2 + 1) < *it2) {
        std::iter_swap(it2, it2 + 1);
      }
    }
  }
}

int main() {
  int input[] = {1, 45, 756, 4569, 56, 3, 8, 5, -10, -4};
  
  //before sorting
  std::cout << "Unsorted array:\n";
  print_range(std::begin(input), std::end(input));
   
   //calling bubble_sort function
  bubble_sort(std::begin(input), std::end(input));
  
  //after sorting
  std::cout << "\nSorted array:\n";
  print_range(std::begin(input), std::end(input));
}

File: contents/bubble_sort/code/visualbasic/bubble_sort.vb
Imports System

Module Module1
    Public Sub Print_range(Input_array() As Double)
        For Each element In Input_array
            Console.WriteLine(element)
        Next element
    End Sub

    Public Sub Bubble_sort(ByRef Input_array() As Double)
        For it1 = 0 To Input_array.Length
            For it2 = 1 To Input_array.Length - 1
                'these are unsorted! gonna swap 'em
                If (Input_array(it2) < Input_array(it2 - 1)) Then
                    'Swaping
                    Dim temp As Double
                    temp = Input_array(it2)
                    Input_array(it2) = Input_array(it2 - 1)
                    Input_array(it2 - 1) = temp
                End If
            Next it2
        Next it1
    End Sub

    Public Sub Main()
        Dim input() As Double = {1.0, 3, 2, 4, 5, 10, 50, 7, 1.5, 0.3}
        'Using 1.0 since 1. is unable to read
        Console.WriteLine("Before Sorting:")
        Print_range(input)

        Bubble_sort(input)

        Console.WriteLine("After Sorting:")
        Print_range(input)
    End Sub
End Module
